packages/nimble-components/src/table/index.ts

```diff
@@ -323,6 +355,11 @@ export class Table<
         this.tableValidator.validateColumnSortIndices(
             this.getColumnsParticipatingInSorting().map(x => x.sortIndex!)
         );
+        this.tableValidator.validateColumnGroupIndices(
+            this.columns
```

Comment:
Maybe there should be a function similar to `getColumnsParticipatingInSorting()` but for grouping. Right now this validation code is inconsistent with the code within `calculateTanStackGroupingState()` because one looks at `internalGroupingDisabled` and the other does not.